src/handler.ts

```diff
@@ -131,35 +131,41 @@ async function handleRequest(event: FetchEvent): Promise<Response> {
     currentHost = request.headers.get('host') || request.headers.get('hostname')
   }
 
+  let url: string = request.url
   let response: Response
-  if (request.url.includes('forward=true') == true) {
+
+  // If the request contains a 'x-original-url' header we understand that this request is forwarded
+  // to the worker and that therefor the upstream should not be fetched. We use a custom header to
+  // change as little as possible from the original request.
+  if (request.headers.get('x-original-url') != null) {
     response = new Response('ok', { status: 200 })
+    url = request.headers.get('x-original-url') || request.url
```

Comment:
wait a sec, so `request.headers.get` returns `false` if the header doesn't exist? If so, why not only use `url = request.headers.get('x-original-url') || request.url` and skip the `if (request.headers.get('x-original-url') != null) {` line altogether? Only in the case where `x-original-url` is set will it be overwritten. Or am I misunderstanding the logic here?

Reply:
`request.headers.get('x-original-url')` returns `null|string` but it can be used in this short or syntax. In any case I can definitively move the `url` initialization up like you mentioned:

```typescript
let url = request.headers.get('x-original-url') || request.url
```

but we still need the conditional for creating the new `Response` instance. On the plus side it makes easier the code.